keyboards/keskinpos/keskinpos75/keymaps/custom/keymap.c

```diff
@@ -0,0 +1,345 @@
+       MACRO13, MACRO14,  MACRO15, MACRO16,  MACRO17,   MACRO18,     _______,         KC_F12,      KC_F12,       _______,    _______,   _______,    KC_PAST,   _______,   _______,
+       MACRO19, MACRO20,  MACRO21, MACRO22,  MACRO23,   MACRO24,    _______,         _______,     KC_UP,        KC_ENT,     KC_P7,     KC_P8,      KC_P9,     KC_F9,     KC_F9,
+       MACRO25, MACRO26,  MACRO27, MACRO28,  MACRO29,   MACRO30,      _______,         _______,     KC_DOWN,      _______,    KC_P4,     KC_P5,      KC_P6,     KC_F9,     KC_F9,
+       MACRO31, MACRO32,  MACRO33, MACRO34,  MACRO35,   MACRO36,    KC_F3,           _______,     KC_BSPC,      KC_0,       KC_P1,     KC_P2,      KC_P3,     KC_F5,     KC_F5,
+       MACRO37, MACRO38,  MACRO39, MACRO40,  MACRO41,   MACRO42,    _______,         _______,     _______,      KC_0,       KC_P0,     TR_DOT,     KC_P0,     KC_F5,     KC_F5
+),
+       [2] = LAYOUT_ortho_7x15(
+        KC_P3,    KC_B,    KC_C,    TR_CCED,    KC_D,    KC_E,      KC_NUMLOCK,      KC_F1,       KC_F1,        KC_F4,      KC_F4,     KC_F10,      KC_F10,   _______,   _______,
+        KC_F,    KC_G,    TR_GBRV, KC_H,       KC_I,    TR_IDOT,   KC_CAPSLOCK,     KC_F2,       KC_F2,        KC_F6,      KC_F6,     _______,    _______,   _______,   _______,
+        KC_J,    KC_K,    KC_L,    KC_M,       KC_N,    KC_O,      _______,         KC_F12,      KC_F12,       _______,    _______,   _______,    KC_PAST,   _______,   _______,
+        TR_ODIA, KC_P,    KC_Q,    KC_R,       KC_S,    TR_SCED,   _______,         _______,     KC_UP,        KC_ENT,     KC_P7,     KC_P8,      KC_P9,     KC_F9,     KC_F9,
+        KC_T,    KC_U,    TR_UDIA, KC_V,       KC_W,    KC_X,      _______,         _______,     KC_DOWN,      _______,    KC_P4,     KC_P5,      KC_P6,     KC_F9,     KC_F9,
+        KC_Y,    KC_Z,    KC_SPC,  TR_LPRN,    TR_RPRN, TR_UNDS,   KC_F3,           _______,     KC_BSPC,      KC_0,       KC_P1,     KC_P2,      KC_P3,     KC_F5,     KC_F5,
+        TR_DOT,  TR_COMM, KC_PSLS, TR_COLN,    TR_EQL,  TR_ASTR,   _______,         _______,     _______,      KC_0,       KC_P0,     TR_DOT,     KC_P0,     KC_F5,     KC_F5
+)
+
+
+};
+
+void dip_switch_update_user(uint8_t index, bool active) {
+	switch (index) {
+		case 0:
+		if(active) { layer_on(1); } else { layer_off(1); }
+		break;
+		case 1:
+		if(active) { layer_on(2); } else { layer_off(2); }
+		break;
+        case 2:
+        if(active) { layer_on(3); } else { layer_off(3); }
+        break;
+	}
+}
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch(keycode) {
+            case MACRO01:
+                register_code(KC_LCTRL);
+                register_code(KC_S);
+                unregister_code(KC_S);
+                unregister_code(KC_LCTRL);
```

Comment:
You can replace this with `C(KC_S)` in your keymap. No need for a macro for this.